drivers/ma_yya_report/app/models/ma_yya_report/universe_calculator.rb

```diff
@@ -196,19 +189,56 @@ def calculate(&post_processor)
         map(&:CurrentLivingSituation)
     end
 
+    # Most recent Zip for each source client
     private def zip_codes(client)
-      client.source_hmis_clients.map(&:processed_youth_current_zip).compact.uniq || []
+      hmis_source_clients_for(client).map do |hmis_client|
+        hmis_client.addresses.sort(:DateUpdated).pluck(:postal_code).compact.last
+      end.uniq
     end
 
+    # All flex funds received by all source clients (e.g. ["Transportation", "Child care", "Move-in"])
+    # NOTE: Assumes that there is only 1 HMIS data source
     private def flex_funds(client)
-      client.source_hmis_forms.
-        within_range(@filter.start_date .. @filter.end_date).
-        where(name: 'Flex Funds').
-        map(&:flex_funds).flatten.uniq
+      return [] unless flex_funds_service_type && flex_funds_types_cded
+
+      hmis_personal_ids = hmis_source_clients_for(client).pluck(:personal_id)
+      # IDs for all Flex Fund services rendered to the client(s)
+      flex_funds_service_ids = Hmis::Hud::CustomService.hmis.where(personal_id: hmis_personal_ids, service_type: flex_funds_service_type).pluck(:id)
+      # CustomDataElements tied to these CustomServices indicate which fund types were provided
+      flex_funds_types_cded.values.
+        where(owner_id: flex_funds_service_ids).
+        pluck(:value_string).uniq.
+        # Remove parenthesized text from "Move-in (Including first, last and security)"
+        map { |ff_type| ff_type.gsub(/\([^()]*\)/, '').strip }.uniq
```

Comment:
This is missing the date range limitation `within_range(@filter.start_date .. @filter.end_date).` (we should only be looking at flex funds within the report dates, and I _think_ we can save at least one query, by making that a `select` instead of `pluck` so it becomes a subquery.  They are in the same database so that _should_ be safe.